xarray/backends/api.py

```diff
@@ -6,6 +6,11 @@
 from glob import glob
 from io import BytesIO
 from numbers import Number
+try:
```

Comment:
This prefers `pathlib2` and only falls back to `pathlib` if necessary. I'm following <https://github.com/mcmtroffaes/pathlib2/issues/8#issuecomment-162890916> here although preferring the stdlib module would feel better.

What do you think?